src/components/Reports/ReportsBox/ReportsBox.js

```diff
@@ -37,7 +37,12 @@ class ReportsBox extends React.Component {
   }
 
   componentWillReceiveProps(nextProps) {
```

Comment:
Maybe it is not a good idea using this function, [this article](https://overreacted.io/writing-resilient-components) talks a lot about it. 

I think that they also [propose some ways to fix it](https://overreacted.io/writing-resilient-components/#dont-stop-the-data-flow-in-side-effects): 

```js
class SearchResults extends React.Component {
  state = {
    data: null,
    currentPage: 0,
  };
  componentDidMount() {
    this.fetchResults();
  }
  componentDidUpdate(prevProps, prevState) {
    if (
      prevState.currentPage !== this.state.currentPage || // ✅ Refetch on change      prevProps.query !== this.props.query
    ) {
      this.fetchResults();
    }
  }
  fetchResults() {
    const url = this.getFetchUrl();
    // Do the fetching...
  }
  getFetchUrl() {
    return (
      'http://myapi/results?query' + this.props.query +
      '&page=' + this.state.currentPage // ✅ Updates are handled    );
  }
  render() {
    // ...
  }
}
```

It has a flaw: 

> **Wouldn’t it be nice if we could somehow automatically catch these mistakes?** Isn’t that something a linter could help us with?
>
> [ . . . ]
>
> Unfortunately, automatically checking a class component for consistency is too difficult.
> 

After that, they give a nice example using `useEffec` hook. It is very very interesting and maybe we should follow that approach.

Anyway, I think that using hooks is only a nice to have, but I would like to use `componentDidUpdate` in place of `componentWillReceiveProps`, what do you think?

Reply:
I found this [specific scenario with the suggested solution](https://reactjs.org/blog/2018/03/27/update-on-async-rendering.html#fetching-external-data-when-props-change):

> The recommended upgrade path for this component is to move data updates into componentDidUpdate. You can also use the new getDerivedStateFromProps lifecycle to clear stale data before rendering the new props: [ . . . ]

Reply:
Update pr with this changes